packages/pluginutils/src/dataToEsm.ts

```diff
@@ -0,0 +1,96 @@
+import { DataToEsm } from '../types';
+
+import makeLegalIdentifier from './makeLegalIdentifier';
+
+export type Indent = string | null | undefined;
+
+function stringify(obj: any): string {
+  return (JSON.stringify(obj) || 'undefined').replace(
+    /[\u2028\u2029]/g,
+    (char) => `\\u${`000${char.charCodeAt(0).toString(16)}`.slice(-4)}`
+  );
+}
+
+function serializeArray<T>(arr: Array<T>, indent: Indent, baseIndent: string): string {
+  let output = '[';
+  const separator = indent ? `\n${baseIndent}${indent}` : '';
+  for (let i = 0; i < arr.length; i++) {
+    const key = arr[i];
+    output += `${i > 0 ? ',' : ''}${separator}${serialize(key, indent, baseIndent + indent)}`;
+  }
+  return `${output}${indent ? `\n${baseIndent}` : ''}]`;
+}
+
+function serializeObject<T>(obj: { [key: string]: T }, indent: Indent, baseIndent: string): string {
+  let output = '{';
+  const separator = indent ? `\n${baseIndent}${indent}` : '';
+  const keys = Object.keys(obj);
+  for (let i = 0; i < keys.length; i++) {
+    const key = keys[i];
+    const stringKey = makeLegalIdentifier(key) === key ? key : stringify(key);
+    output += `${i > 0 ? ',' : ''}${separator}${stringKey}:${indent ? ' ' : ''}${serialize(
+      obj[key],
+      indent,
+      baseIndent + indent
+    )}`;
+  }
+  return `${output}${indent ? `\n${baseIndent}` : ''}}`;
+}
+
+function serialize(obj: any, indent: Indent, baseIndent: string): string {
```

Comment:
It would be good to prefer `unknown` over `any` since it has better type safety.

Reply:
Let's tackle overall TS best practices post-migration. We're looking for an analog to the current version right now.